src/relayclient/RelayClient.ts

```diff
@@ -162,15 +164,25 @@ export class RelayClient {
     return false
   }
 
-  async _init (): Promise<void> {
-    if (this.initialized) { return }
-    this.emit(new GsnInitEvent())
-    await this.contractInteractor.init()
-    this.initialized = true
+  async init (): Promise<this> {
+    await this.initMutex.runExclusive(async () => {
+      if (this.initialized) {
```

Comment:
Obviously, it would be much simpler not to have a mutex and only support one way to init an object, wouldn't it?

Reply:
one way would mean you MUST wait for complete initialization before calling "send".
this way, you can init in the background, and call send even before its finished. the send will wait for init to complete and continue.